Source/WebCore/platform/graphics/SurrogatePairAwareTextIterator.h

```diff
@@ -29,8 +29,8 @@ class SurrogatePairAwareTextIterator {
 public:
     // The passed in UChar pointer starts at 'currentIndex'. The iterator operates on the range [currentIndex, lastIndex].
     // 'endIndex' denotes the maximum length of the UChar array, which might exceed 'lastIndex'.
-    SurrogatePairAwareTextIterator(const UChar* characters, unsigned currentIndex, unsigned lastIndex, unsigned endIndex)
-        : m_characters(characters)
+    SurrogatePairAwareTextIterator(std::span<const UChar> characters, unsigned currentIndex, unsigned lastIndex, unsigned endIndex)
```

Comment:
We should use `characters.size() - currentIndex` as `endIndex`, right?

Reply:
Seems like this would be a behavior change?

Reply:
I don’t think so, no. I looked at all the call sites. Can you find one where it’s not correct?

Reply:
This one is WidthIterator?
```
SurrogatePairAwareTextIterator textIterator(m_run->subspan16(m_currentCharacterIndex), m_currentCharacterIndex, offset, length);
```

As far as I can tell, `length == m_run->span16().size()` but `m_currentCharacterIndex` is not necessarily 0?

Reply:
Oh never mind, I'm tired. `m_run->subspan16(m_currentCharacterIndex).size() == m_run->span16().size() - m_currentCharacterIndex`. I'll make the change.

Reply:
But then, shouldn't we use `endIndex = characters.size() + currentIndex` instead of `endIndex = characters.size() - currentIndex`?

Reply:
Yes, I had it wrong.